src/vm/ilmarshalers.cpp

```diff
@@ -4669,6 +4680,8 @@ FCIMPL3(void, MngdFixedArrayMarshaler::ConvertSpaceToManaged, MngdFixedArrayMars
     //
     SetObjectReference(pManagedHome, AllocateSzArray(pThis->m_Array, pThis->m_cElements));
```

Comment:
This is potential GC hole. It is valid for C++ compiler to fetch the value of `pManagedHome`, then call AllocateSzArray - that will move the object that pManagedHome points to, and then call SetObjectReference - with the bogus value of pManagedHome.

This needs to be split into multiple lines